### PR TITLE
Bug 2000846: Structure the logs to same format in the entire codebase 

### PR DESCRIPTION
**Pull request by priyanka19-98**
Signed off by: Priyanka Jiandani <pjiandan@redhat.com>

### CODE REVIEW

**Comment by gnufied** [comment]
Can you post a sample of logs generated after the change?

**Comment by priyanka19-98** [comment]
LSO logs: 
```2021-08-10T07:00:23.009Z	INFO	setup	Go Version: go1.16.6
2021-08-10T07:00:23.009Z	INFO	setup	Go OS/Arch: linux/amd64
2021-08-10T07:00:23.009Z	INFO	setup	local-storage-operator Version: 0aed82783ce49f8de41b4e181dab7365cc26327f
2021-08-10T07:00:23.613Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2021-08-10T07:00:23.614Z	INFO	setup	starting manager
I0810 07:00:23.614597       1 leaderelection.go:243] attempting to acquire leader lease default/98d5776d.storage.openshift.io...
2021-08-10T07:00:23.615Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I0810 07:00:39.142826       1 leaderelection.go:253] successfully acquired lease default/98d5776d.storage.openshift.io
2021-08-10T07:00:39.143Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"default","name":"98d5776d.storage.openshift.io","uid":"a586820e-db91-4fa5-b3f7-e4ac44c29b25","apiVersion":"v1","resourceVersion":"275703"}, "reason": "LeaderElection", "message": "local-storage-operator-564c57d75c-glctl_5df2e36e-0f4f-4651-95eb-d52139df71a5 became leader"}
2021-08-10T07:00:39.143Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"Lease","namespace":"default","name":"98d5776d.storage.openshift.io","uid":"ed10165f-010b-4e0f-b7ba-c93cd9b261cd","apiVersion":"coordination.k8s.io/v1","resourceVersion":"275704"}, "reason": "LeaderElection", "message": "local-storage-operator-564c57d75c-glctl_5df2e36e-0f4f-4651-95eb-d52139df71a5 became leader"}
2021-08-10T07:00:39.143Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.143Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.143Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting Controller	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume"}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolume	Starting Controller	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume"}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolumediscovery	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeDiscovery", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.144Z	INFO	controller-runtime.manager.controller.localvolumediscovery	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeDiscovery", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumediscovery	Starting Controller	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeDiscovery"}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet", "source": "kind source: /, Kind="}
2021-08-10T07:00:39.145Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting Controller	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet"}
2021-08-10T07:00:39.246Z	INFO	controller-runtime.manager.controller.localvolumediscovery	Starting workers	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeDiscovery", "worker count": 1}
2021-08-10T07:00:39.246Z	INFO	controller-runtime.manager.controller.localvolume	Starting workers	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "worker count": 1}
2021-08-10T07:00:39.247Z	INFO	controller-runtime.manager.controller.localvolume	Starting workers	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "worker count": 1}
2021-08-10T07:00:39.247Z	INFO	controller-runtime.manager.controller.localvolumeset	Starting workers	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolumeSet", "worker count": 1}
```

**Comment by gnufied** [comment]
I personally think that missing source location is a big problem with these logs. Please see corresponding issue in kubernetes - https://github.com/kubernetes/kubernetes/issues/102353

Until we can solve source location problem, we should either move everything to klog or fix source location in new structured logs.

**Comment by priyanka19-98** [comment]
After modifying log config: 
```
2021-08-10T19:56:21.177Z	INFO	setup	local-storage-operator/main.go:64	Go Version: go1.16.6
2021-08-10T19:56:21.177Z	INFO	setup	local-storage-operator/main.go:65	Go OS/Arch: linux/amd64
2021-08-10T19:56:21.177Z	INFO	setup	local-storage-operator/main.go:66	local-storage-operator Version: 67b5b83c46e4adb5e933ba7498dbbdded094ba12
2021-08-10T19:56:21.798Z	INFO	controller-runtime.metrics	metrics/listener.go:44	metrics server is starting to listen	{"addr": ":8080"}
2021-08-10T19:56:21.803Z	INFO	setup	local-storage-operator/main.go:156	starting manager
I0810 19:56:21.803534       1 leaderelection.go:243] attempting to acquire leader lease default/98d5776d.storage.openshift.io...
2021-08-10T19:56:21.804Z	INFO	controller-runtime.manager	manager/manager.go:290	starting metrics server	{"path": "/metrics"}
I0810 19:56:37.421011       1 leaderelection.go:253] successfully acquired lease default/98d5776d.storage.openshift.io
2021-08-10T19:56:37.421Z	DEBUG	controller-runtime.manager.events	record/event.go:308	Normal	{"object": {"kind":"ConfigMap","namespace":"default","name":"98d5776d.storage.openshift.io","uid":"a586820e-db91-4fa5-b3f7-e4ac44c29b25","apiVersion":"v1","resourceVersion":"332558"}, "reason": "LeaderElection", "message": "local-storage-operator-6b4bdd7869-qgj99_ce2a6f16-5cf2-4c7f-9a93-72552355b364 became leader"}
2021-08-10T19:56:37.424Z	DEBUG	controller-runtime.manager.events	record/event.go:308	Normal	{"object": {"kind":"Lease","namespace":"default","name":"98d5776d.storage.openshift.io","uid":"ed10165f-010b-4e0f-b7ba-c93cd9b261cd","apiVersion":"coordination.k8s.io/v1","resourceVersion":"332559"}, "reason": "LeaderElection", "message": "local-storage-operator-6b4bdd7869-qgj99_ce2a6f16-5cf2-4c7f-9a93-72552355b364 became leader"}
2021-08-10T19:56:37.423Z	INFO	controller-runtime.manager.controller.localvolume	controller/controller.go:221	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T19:56:37.424Z	INFO	controller-runtime.manager.controller.localvolume	controller/controller.go:221	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
2021-08-10T19:56:37.424Z	INFO	controller-runtime.manager.controller.localvolume	controller/controller.go:221	Starting EventSource	{"reconciler group": "local.storage.openshift.io", "reconciler kind": "LocalVolume", "source": "kind source: /, Kind="}
```

**Comment by gnufied** [comment]
Can you link the commit that changed the log config? I think you rebased - so I can't see exact diff that introduced the change.

File: main.go [hunk 1]
```diff
@@ -69,14 +70,17 @@ func main() {
...
 	var probeAddr string
+
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
 	opts := zap.Options{
 		Development: true,
+		ZapOpts:     []zaplog.Option{zaplog.AddCaller()},
```
Comment (priyanka19-98) [inline main.go:81]: @gnufied  here the change was introduced. I pushed the changes today.

File: diskmaker_manager/manager.go [hunk 2]
```diff
@@ -41,6 +42,7 @@ func startManager(cmd *cobra.Command, args []string) error {
 	klog.InitFlags(klogFlags)
 	opts := zap.Options{
 		Development: true,
+		ZapOpts:     []zaplog.Option{zaplog.AddCaller()},
```
Comment (priyanka19-98) [inline diskmaker_manager/manager.go:45]: Also here

**Comment by gnufied** [comment]
@priyanka19-98 the PR will need BZ for merging though.

**Comment by priyanka19-98** [comment]
> @priyanka19-98 the PR will need BZ for merging though.

@gnufied  should I create a BZ, or we want to wait?

**Comment by gnufied** [comment]
@priyanka19-98 yeah lets create a BZ that targets 4.9.0. Otherwise, it will be weeks before the PR can merge.

**Comment by priyanka19-98** [comment]
@gnufied  could you approve it again? I had to rebase it against some latest changes.

**Comment by gnufied** [comment]
@priyanka19-98 I am thinking that - lets merge this PR for 4.10. It is not super urgent to get in today and I am slightly worried if this causes any regressions which we may not catch in time.

**Pull request merged**